geode-core/src/test/java/org/apache/geode/internal/statistics/LinuxSystemStatsTest.java

```diff
@@ -0,0 +1,189 @@
+  private File testDir;
+
+  @Before
+  public void setUp() throws Exception {
+    this.testDir = this.temporaryFolder.getRoot();
+    assertTrue(this.testDir.exists());
+    System.setProperty(SimpleStatSampler.ARCHIVE_FILE_NAME_PROPERTY, this.testDir.getAbsolutePath()
+        + File.separator + SimpleStatSampler.DEFAULT_ARCHIVE_FILE_NAME);
+    LinuxProcFsStatistics.init();
+    initStats();
+    StatisticsTypeImpl statisticsType = (StatisticsTypeImpl) LinuxSystemStats.getType();
+    LocalStatisticsImpl statistics = (LocalStatisticsImpl) getStatisticsManager()
+        .createStatistics(statisticsType, statisticsType.getName());
+
+    ints = statistics._getIntStorage();
+    longs = statistics._getLongStorage();
+    doubles = statistics._getDoubleStorage();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    StatisticsTypeFactoryImpl.clear();
+    if (this.statisticsFactory != null) {
+      this.statisticsFactory.close();
+      this.statisticsFactory = null;
+    }
+  }
+
+  @Test
+  public void testFlatStealTime() throws Exception {
+    String[] results = {"cpu  0 0 0 0 0 0 0 0 0 0",
+        // add on 4 clicks 4 idle 0 steal
+        "cpu  0 0 0 4 0 0 0 0 0 0"};
+
+    doTest(results, 0);
+  }
+
+  @Test
+  public void test25PercentStealTime() throws Exception {
+
+    String[] results = {"cpu  0 0 0 0 0 0 0 0 0 0",
+        // add on 4 clicks 3 idle 1 steal
+        "cpu  0 0 0 3 0 0 0 1 0 0"};
+
+    doTest(results, 25);
+  }
+
+  @Test
+  public void test66PercentStealTime() throws Exception {
+
+    String[] results = {"cpu  0 0 0 0 0 0 0 0 0 0",
+        // add on 3 clicks 1 idle 2 steal
+        "cpu  0 0 0 1 0 0 0 2 0 0"};
+
+    doTest(results, 66);
+  }
+
+  @Test
+  public void test100PercentStealTime() throws Exception {
+
+    String[] results = {"cpu  0 0 0 0 0 0 0 0 0 0",
+        // add on 1 clicks 0 idle 1 steal
+        "cpu  0 0 0 0 0 0 0 1 0 0"};
+    doTest(results, 100);
+  }
+
+  protected void doTest(String[] results, int expectedStatValue) throws Exception {
+
+    Answer<FileInputStream> answer = new MyStealTimeAnswer(results);
+    PowerMockito.whenNew(FileInputStream.class).withArguments(anyString()).thenAnswer(answer);
+
+
+    LinuxProcFsStatistics.refreshSystem(ints, longs, doubles);
+    LinuxProcFsStatistics.refreshSystem(ints, longs, doubles);
+
+    Statistics[] statistics = getStatisticsManager().findStatisticsByTextId("LinuxSystemStats");
+    waitForExpectedStatValue(statistics[0], "cpuSteal", expectedStatValue, 20000, 10);
+  }
+
+  protected void initStats() {
+    statisticsFactory = new LocalStatisticsFactory(new CancelCriterion() {
+      public String cancelInProgress() {
+        return null;
+      }
+
+      public RuntimeException generateCancelledException(Throwable e) {
+        return null;
+      }
+    });
+  }
+
+  private File writeStringToFile(String string) throws IOException {
+    File file = File.createTempFile("LinuxSystemStatsTest", ".test");
```

Comment:
I'm not real sure why you need a file, but if you do, then you should probably use the TemporaryFolder instead of `File.createTempFile` for something like this:
```java
File file = new File(temporaryFolder.getRoot(), "LinuxSystemStatsTest.test");
```
And then delete the line `file.deleteOnExit();` because TemporaryFolder will delete everything you written to under its root folder.

Looks like something else would also need to change because if I try to change your test to use temporaryFolder, 3 of the tests fail -- I'm not sure what it's running into.